src/content/en/updates/2018/07/css-scroll-snap.md

```diff
@@ -0,0 +1,365 @@
+effective box similar to how `scroll-padding` functions on snap scroll
+container.
+
+You may have noticed that these two properties do not have the word "`snap`" in
+them. This is intentional as they actually modify the box for all relevant
+scroll operations and are not just scroll snapping. For example Chrome takes
+them into account when calculating page size for paging scroll operations such
+as PageDown and PageUp and also when calculating scroll amount for
+`Element.scrollIntoView()` operation.
+
+
+<figure>
+  <a href="/web/updates/images/2018/07/css-scroll-snap/product-page.mp4"
+     target="_blank">
+    <video src="/web/updates/images/2018/07/css-scroll-snap/product-page.mp4"
+           autoplay muted playsinline loop controls alt="Product page demo" height="600"></video>
+  </a>
+  <figcaption>
+    <a href="https://snap.glitch.me/product.html"
+       target="_blank">View demo</a> |
+    <a href="https://glitch.com/edit/#!/snap?path=product.html:1:0"
+       target="_blank">Source</a>
+  </figcaption>
+</figure>
+
+## Interaction with other scrolling APIs
+
+### DOM Scrolling API {#dom-scrolling-api}
+
+Scroll snapping happens **after** all scroll operations including those
+initiated by script. When you are using APIs like `Element.scrollTo`, the
+browser will calculate the intended scroll position of the operation, then apply
+appropriate snapping logic to find the final snapped location. Thus,  there is
+no need for user script to do any manual calculations for snapping.
+
+
+### Smooth Scrolling  {#smooth-scrolling}
+
+Smooth scrolling controls the behavior of a programmatic scroll operation while
+scroll snap determines its destination. Since they control orthogonal aspects of
+scrolling, they can be used together and complement each other.
+
+
+### Overscroll Behavior {#overscroll-behavior}
+
+[Overscroll behavior
+API](https://developers.google.com/web/updates/2017/11/overscroll-behavior)
+controls how scroll is chained across multiple elements and it is not affected
+by scroll snap.
+
+
+## Caveats and best practices
+
+
+Avoid using mandatory snapping when target elements are widely spaced apart.
+This can cause content in between the snap positions to become inaccessible.
+
+Use `CSS.supports` for feature detecting CSS Scroll Snap.  But avoid using
+`scroll-snap-type` which is also present in the deprecated specification and can
+be unreliable.
+
+
+```
+if (CSS.supports('scroll-snap-align: start')) {
+  // use css scroll snap
+} else {
+  // use fallback
+}
+```
+
+
+Do not assume that programmatically scrolling APIs such as `Element.scrollTo`
+always finish at the requested scroll offset. Scroll snapping may adjust the
+scroll offset after programmatic scrolling is complete. Note that this was not a
+good assumption even before scroll snap since scrolling may have been
+interrupted for other reasons but it is especially the case with scroll
+snapping.
+
+
+<aside>
+  <strong>Note:</strong>
```

Comment:
ProTip: You can use [markdown for callouts](https://developers.google.com/internal/publishing/reference/styles#callouts_notes_etc) for example:

```
Note: There is an [upcoming](#link) proposal to change various...
```

You don't have to change, just a tip for next time.

Reply:
Neat! Changed here as well.